Update wording to include prop behaviour (#4460)
sveltejs/svelte: site/content/docs/01-component-format.md

---

> You can specify an initial value for a prop. It will be used if the component's consumer doesn't specify the prop on the component. Note that whenever a specified prop is removed by the consumer, the value is set to `undefined` rather than the initial value.

Reviewer: what about
```suggestion
You can specify an initial value for a prop. It will be used if the component's consumer doesn't specify the prop when instancing the component. Note that whenever a specified prop is removed by the consumer, the value is set to `undefined` rather than the initial value.
```
?

Reviewer 2: I think the right term really is "default initial value". "initial value" alone implies that this will be the starting value, unconditionally.

Furthermore, "if the component's consumer doesn't specify **the prop on the component**" is slightly misleading because `<Cmp x={undefined} />` also results in the default value being used. The following would be more accurate: "if the initial value of the prop is undefined".

That being said, "You can specify a default initial value for a prop." says it all, as far as I'm concerned. The next sentence seems a little superfluous. The real important notice, about which there has been some confusion, is the last one.

Reviewer: @rixo you are not wrong, hopefully this would be clear enough?

```suggestion
You can specify a default initial value for a prop. It will be used if the component's consumer doesn't specify the prop on the component (or if its initial value is `undefined`). Note that whenever a prop is removed by the consumer, its value is set to `undefined` rather than the initial value.
```

---

> You can specify a default value for a prop. It will be used if the component's consumer doesn't specify the prop on the component. Note that whenever a specified prop is removed by the consumer, the value is set to `undefined` rather than the default value.

Reviewer: Given the current behaviour I'd prefer we don't use the term `default` anymore as it can be misleading.
